cmd/influx_inspect/verify_seriesfile/verify_seriesfile.go

```diff
@@ -0,0 +1,298 @@
+// Package verify_seriesfile verifies integrity of series files.
+package verify_seriesfile
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"github.com/influxdata/influxdb/logger"
+	"github.com/influxdata/influxdb/tsdb"
+	"go.uber.org/zap"
+)
+
+// Command represents the program execution for "influx_inspect verify-seriesfile".
+type Command struct {
+	Stderr io.Writer
+	Stdout io.Writer
+}
+
+// NewCommand returns a new instance of Command.
+func NewCommand() *Command {
+	return &Command{
+		Stderr: os.Stderr,
+		Stdout: os.Stdout,
+	}
+}
+
+// Run executes the command.
+func (cmd *Command) Run(args ...string) error {
+	fs := flag.NewFlagSet("verify-seriesfile", flag.ExitOnError)
+	fs.SetOutput(cmd.Stdout)
+
+	dataDir := fs.String("dir", os.Getenv("HOME")+"/.influxdb", "data directory.")
+	dbPath := fs.String("db", "", "database path. overrides data directory.")
+	filePath := fs.String("file", "", "series file path. overrides db and data directory.")
+
+	if err := fs.Parse(args); err != nil {
+		return err
+	}
+
+	return cmd.run(*dataDir, *dbPath, *filePath)
+}
+
+// run performs all of the validations, preferring a filePath, then a dbPath,
+// and then a dataDir.
+func (cmd *Command) run(dataDir, dbPath, filePath string) error {
+	if filePath != "" {
+		_, err := cmd.runFile(filePath)
+		return err
+	}
+	if dbPath != "" {
+		_, err := cmd.runFile(filepath.Join(dbPath, "_series"))
+		return err
+	}
+
+	dbs, err := ioutil.ReadDir(filepath.Join(dataDir, "data"))
+	if err != nil {
+		return err
+	}
+
+	for _, db := range dbs {
+		if !db.IsDir() {
+			continue
+		}
+		_, err := cmd.runFile(filepath.Join(dataDir, "data", db.Name(), "_series"))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// runFile performs validations on a series file. The error is only returned if
+// there was some fatal problem with operating, not if there was a problem with
+// the series file.
+func (cmd *Command) runFile(filePath string) (valid bool, err error) {
+	logger := logger.New(cmd.Stderr).With(zap.String("path", filePath))
```

Comment:
nit: this could be set higher up, and then this line could be: `logger := cmd.logger.With(zap.String("path", filePath))`